yaml.go

```diff
@@ -23,9 +23,41 @@ import (
 
 // ParseYAML parses yaml-encoded byte string into the struct
 // passed to the function.
-func ParseYAML(data []byte, cfg interface{}) error {
+// EnableTemplating() argument allows to treat configuration file as a template
+// for example, it will support {{env "VAR"}} - that will substitute
+// environment variable "VAR" and pass it to YAML file parser
+func ParseYAML(data []byte, cfg interface{}, funcArgs ...ParseOption) error {
```

Comment:
I don't want to enable templating by default, so it will be there as a functional argument only when necessary, what makes a good use of functional arguments in this particular case.